### PR TITLE
fix handling of invalid tags in artifactory

### PR DESCRIPTION
**Pull request by redhatrises**

### CODE REVIEW

File: pkg/registry/falcon_registry/registry.go [hunk 1]
```diff
@@ -117,6 +121,57 @@ func guessLastTag(tags []string, filter func(string) bool) (string, error) {
 func listDockerTags(ctx context.Context, sys *types.SystemContext, imgRef types.ImageReference) ([]string, error) {
 	tags, err := docker.GetRepositoryTags(ctx, sys, imgRef)
 	if err != nil {
```
Comment (cs-alphonse) [inline pkg/registry/falcon_registry/registry.go:123]: Should we check for a specific `err` and then execute the `http` request?
Or is that handled indirectly by checking `len(tags)` below.
IIUC, partial list of tags populated in `tags` also is considered an error.

File: pkg/registry/falcon_registry/registry.go [hunk 2]
```diff
@@ -117,6 +121,57 @@ func guessLastTag(tags []string, filter func(string) bool) (string, error) {
...
+				return nil, fmt.Errorf("Unable to get http response to list container registry tags: %v", err)
+			}
+			defer resp.Body.Close()
+
+			bodyText, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("Unable to read response body to list registry tags: %v", err)
+			}
+
+			if err := json.Unmarshal(bodyText, &jsonRes); err != nil {
```
Comment (cs-alphonse) [inline pkg/registry/falcon_registry/registry.go:156]: Is the response from our API always in `JSON` format? 
(i,e) do we need to pass format in our request?

File: pkg/registry/falcon_registry/registry.go [hunk 3]
```diff
@@ -117,6 +121,57 @@ func guessLastTag(tags []string, filter func(string) bool) (string, error) {
 func listDockerTags(ctx context.Context, sys *types.SystemContext, imgRef types.ImageReference) ([]string, error) {
 	tags, err := docker.GetRepositoryTags(ctx, sys, imgRef)
 	if err != nil {
+		// Artifactory incorrectly adds the digest to the tag list, so we need to handle this case
+		// by fetching the tags from the registry directly
+		// This is a workaround, and should be removed once Artifactory is fixed
+		if len(tags) == 0 {
+			var jsonRes map[string]interface{}
+			reg, _, _ := strings.Cut(imgRef.StringWithinTransport(), ":")
```
Comment (cs-alphonse) [inline pkg/registry/falcon_registry/registry.go:129]: We are not checking the return value `found` here.
https://www.codecademy.com/resources/docs/go/strings/cut

File: pkg/registry/falcon_registry/registry.go [hunk 4]
```diff
@@ -117,6 +121,57 @@ func guessLastTag(tags []string, filter func(string) bool) (string, error) {
...
+			reg = strings.Replace(reg, "crowdstrike.com/", "crowdstrike.com/v2/", 1)
```
Comment (cs-alphonse) [inline pkg/registry/falcon_registry/registry.go:130]: For my clarification;
Is it guaranteed that the string `reg` would have the substring `crowdstrike.com/`
(i,e) do we validate that in the caller, before this code is invoked?

File: pkg/registry/falcon_registry/registry.go (hunk 1 above)
Comment (redhatrises) [reply pkg/registry/falcon_registry/registry.go:123]: Looking at the error now for the string `registry returned invalid tag`

File: pkg/registry/falcon_registry/registry.go (hunk 2 above)
Comment (redhatrises) [reply pkg/registry/falcon_registry/registry.go:165]: Yes. A docker registry check is always in json... added it to the header.

**Review by cs-alphonse — APPROVED** [review]

**Pull request merged**